### PR TITLE
Allow to configure the app via web interface

### PR DESCRIPTION
**Pull request by leonklingele**
.. only if config/config.php does not exist.

### CODE REVIEW

**Comment by fancycode** [comment]
How does this work with the packaged version on the Spreedbox? Will the corresponding fields in the UI be still available/editable?

**Comment by leonklingele** [comment]
No changes are required for the packaging.

If `config/config.php` exists, we will use it instead of querying the database.
`OwnCloudConfig.js` is no longer required for the Spreedbox and can simply be deleted.

**Comment by longsleep** [comment]
> If config/config.php exists, we will use it instead of querying the database.

What about the user inferface? Does it detect that the configuration is not in the database?

> OwnCloudConfig.js is no longer required for the Spreedbox and can simply be deleted.

What happens if the file is still there? No problem?

**Comment by leonklingele** [comment]
> What about the user inferface? Does it detect that the configuration is not in the database?

I suggest to test the user interface yourself.

> What happens if the file is still there? No problem?

No problem. We will load it if it exists.

**Comment by leonklingele** [comment]
Sorry, I updated the PR again.
I guess this can be reviewed now.

File: README.md [hunk 1]
```diff
@@ -129,6 +119,11 @@ Follow our [Guide for Apache](./doc/example-config-apache.md) on how to run Spre
 
 Follow our [Guide for Nginx](./doc/example-config-nginx.md) on how to run Spreed WebRTC on the same host as Nextcloud in a Nginx subpath.
 
+## Running Spreed WebRTC and Nextcloud on different origins
+
+Running Spreed WebRTC and Nextcloud on different origins is not recommended, as you will not be able to use the Screensharing feature due to browser restrictions.
+If you still prefer to run them on different origins, please read the [README.md of v0.1.6](https://github.com/strukturag/nextcloud-spreedme/blob/owncloud-spreedme_v016/README.md#preparations).
```
Comment (longsleep) [inline README.md:125]: I think it is a bad to link to a specific version. Can we avoid that somehow?

File: templates/settings-admin.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			<p>
+				<label for="OWNCLOUD_ORIGIN">OWNCLOUD_ORIGIN:</label>
+				<input type="text" id="OWNCLOUD_ORIGIN" name="OWNCLOUD_ORIGIN" placeholder="" readonly="readonly"
+					value="" />
+			</p>
+		</form>
+	</div>
+	<div class="show-if-js-config-missing">
+		<p><code>extra/static/config/OwnCloudConfig.js</code> was not found.</p>
+		<p>If you want to run Spreed WebRTC and Nextcloud on different origins, please read the <a href="https://github.com/strukturag/nextcloud-spreedme/blob/owncloud-spreedme_v016/README.md#preparations" target="_blank">README.md of v0.1.6</a>.</p>
```
Comment (longsleep) [inline templates/settings-admin.php:81]: Also i suggest not to link to a specific version if possible.

**Comment by longsleep** [comment]
Made some comments. Otherwise lgtm :+1 - but please make sure to squash before merge.

File: README.md (hunk 1 above)
Comment (leonklingele) [inline README.md:125]: Well, that was the last version where README.md had a tutorial on how to use different origins.
I can however create a new document in doc/ on how to do so.

**Comment by leonklingele** [comment]
Merged in 7d9843d8ba0ffeb687913e05ca9259cf984d4664.

**Comment by tedd77** [comment]
Installed Spreed-webrtc from repository
1. Should I enable https in /etc/spreed/webrtc.conf 
the server cannot be restarted it fails until removing the https option.
2. I cannot prevent direct access to the spreed server from outside nextcloud
so anyone can access http://serverip:8080
3. Any progress on having the users list of nextcloud displayed on spreed ?

could someone advise ?

**Comment by leonklingele** [comment]
Hi @tedd77,

> 1. Should I enable https in /etc/spreed/webrtc.conf

No, just follow the instructions here: https://github.com/strukturag/nextcloud-spreedme#installation--setup-of-this-app
As you’re using the packaged version of spreed-webrtc, you need to add:

```
[http]
; Full path to directory where to find the server web assets.
root = /usr/share/spreed-webrtc-server/www
```

> 1. I cannot prevent direct access to the spreed server from outside nextcloud so anyone can access http://serverip:8080

This should only happen if you `listen = 0.0.0.0:8080` instead of ´listen = 127.0.0.1:8080`

> 1. Any progress on having the users list of nextcloud displayed on spreed ?

No, but Spreed.ME will be integrated into Nextcloud more deeply with https://github.com/nextcloud/spreed

**Comment by tedd77** [comment]
@leonklingele 
Thanks a lot for your reply.
All is well now, http, https and security no direct access.
however the file /var/www/nextcloud/apps/spreedme/config.php
should have const SPREED_WEBRTC_ORIGIN = ''; (no ip neither port)

also
-----------------/etc/spreed  server.conf----
[http]
; HTTP listener in format ip:port.

## listen = 127.0.0.1:8080

One last request, in order to access it from public ip, I need to know what port on the firewal to open and which protocol. 
Do I need to set Turn server ?

**Comment by tedd77** [comment]
Update Turn is required and port 8443/UDP too

**Comment by leonklingele** [comment]
In order to get https working with Spreed WebRTC, you should set up your web server (most likely nginx or Apache) as a reverse proxy.
Please follow the instructions for [nginx](https://github.com/strukturag/nextcloud-spreedme/blob/master/doc/example-config-nginx.md) or [Apache](https://github.com/strukturag/nextcloud-spreedme/blob/master/doc/example-config-apache.md)

**Comment by tedd77** [comment]
@leonklingele is there a package for Centos 7 without the hassle to compile? if so how to add the repository to the centos machine.

**Comment by leonklingele** [comment]
Currently we only provide packages for Ubuntu: https://github.com/strukturag/spreed-webrtc/wiki/Ubuntu-Repository

**Pull request closed without merging**